mvlearn/semi_supervised/ctregression.py

```diff
@@ -0,0 +1,580 @@
+                    y2.iloc[i] = np.nan
+                    not_null2.pop()
+
+                # appending the index, delta, pred value in list
+                delta2.append([i, delta, pred])
+
+            # sorting the delta1 with descending order of delta value
+            delta2 = sorted(delta2, key=lambda x: -x[1])
+
+            # counts the number of delta value stored
+            count = 0
+
+            # counts the index of delta2
+            index = 0
+
+            while index < len(delta2) and\
+                    count < self.selected_size and\
+                    delta2:
+                # if delta2's delta value is not positive break the loop
+                if delta2[index][1] <= 0:
+                    break
+                else:
+                    count += 1
+                    to_include2.append(delta2[index][0])
+                index += 1
+
+            # removing the unlabelled samples
+            U = [
+                i for i in U
+                if (i not in to_include2)
+                and
+                (i not in to_include1)
+                ]
+            # print("\nto_include1 size {}\tto_include2 size {}"
+            #   .format(len(to_include1), len(to_include2)))
+
+            # if both lists are empty break the iterations
+            if (not to_include1) and (not to_include2):
+                break
+
+            # include the unlablled sample into the
+            # other regressor object training sample
+            not_null1.extend(to_include2)
+            not_null2.extend(to_include1)
+
+            # label the predicted value of each unlabelled sample
+            for i, num in enumerate(to_include1):
+                y2.iloc[num] = delta1[i][2]
+
+            for i, num in enumerate(to_include2):
+                y1.iloc[num] = delta2[i][2]
+
+            # fit the estimator again on the new training set
+            self.estimator1.fit(X1.iloc[not_null1], y1.iloc[not_null1])
+            self.estimator2.fit(X2.iloc[not_null2], y2.iloc[not_null2])
+
+        return self
+
+    def predict(self, Xs):
+        r"""
+        Predict the classes of the examples in the two input views.
+
+        Parameters
+        ----------
+        Xs : list of array-likes or numpy.ndarray
+            - Xs length: n_views
+            - Xs[i] shape: (n_samples, n_features_i)
+            A list of the different views of data to predict.
+
+        Returns
+        -------
+        y_pred : array-like (n_samples,)
+            The predicted value of each input example. The average of
+            the two predicited values is returned
+        """
+
+        X1 = Xs[0]
+        X2 = Xs[1]
+
+        # predicting the value of each view
+        pred1 = self.estimator1.predict(X1)
+        pred2 = self.estimator2.predict(X2)
+
+        # Taking the average of the predicted value and returning it
+        return (pred1 + pred2) * 0.5
+
+    def find_k_nearest_neighbour(self, X, y, index):
```

Comment:
is this supposed to be a function only used within the class? if so, it should have an underscore first, like "_find_k_nearest_neighbor". Also, just for consistency with American english and the rest of the package, can you make it neighbor not neighbour?